src/types/onyx/Policy.ts

```diff
@@ -68,18 +69,115 @@ type TaxRatesWithDefault = {
     taxes: TaxRates;
 };
 
-// These types are for the Integration connections for a policy (eg. Quickbooks, Xero, etc).
-// This data is not yet used in the codebase which is why it is given a very generic type, but the data is being put into Onyx for future use.
-// Once the data is being used, these types should be defined appropriately.
-type ConnectionLastSync = Record<string, unknown>;
-type ConnectionData = Record<string, unknown>;
-type ConnectionConfig = Record<string, unknown>;
-type Connection = {
+type ConnectionLastSync = {
+    successfulDate?: string;
+    errorDate?: string;
+    isSuccessful: boolean;
+    source: 'DIRECT' | 'EXPENSIFYWEB' | 'EXPENSIFYAPI' | 'AUTOSYNC' | 'AUTOAPPROVE';
+};
+
+type Account = {
+    glCode?: string;
+    name: string;
+    currency: string;
+    id: string;
+};
+
+type Employee = {
+    id: string;
+    firstName?: string;
+    lastName?: string;
+    name: string;
+    email: string;
+};
+
+type Vendor = {
+    id: string;
+    name: string;
+    currency: string;
+    email: string;
+};
+
+type TaxCode = {
+    totalTaxRateVal: string;
+    simpleName: string;
+    taxCodeRef: string;
+    taxRateRefs: Record<string, string>;
+    name: string;
+};
+
+type QuickBooksOnlineConnectionData = {
```

Comment:
Why are we making a `data` type specific to QBO but not also a `config` type?

Reply:
config will also differ across accounting software?

Reply:
made the config type specific to QBO

Reply:
> config also differs across accounting software?

Yes, absolutely. The `config` represents how what you want to import data from the accounting software into Expensify, and how you want to export your expenses. All the products we integrate with have differences in how they represent stuff on their end, so each config is going to be different.